code/segmentedtickerplant/datastripe.q

```diff
@@ -22,3 +22,29 @@ initdatastripe:{
      };
 
 if[.ds.datastripe;.proc.addinitlist[(`initdatastripe;`)]];
+
+\d .stpps
+//function to grab where clause from config tables
+segmentfilterfunc:{[tbl;segid] 
+  string first .stpps.segmentfiltermap[exec wcRef from .stpps.segmentconfig where table=tbl , segmentID=segid]};
+// Function to use segmentfilterfunc and return "" incase of error
+segmentfilter:{[tbl;segid]
+  output:.[segmentfilterfunc;
+  (tbl;segid) ;
+  "" ];
+  $[count output>0;output;""]};
```

Comment:
what's the point of this check? seems to me like it wouldn't change the output?

also not sure that the condition is doing what might be expected, assume it should be `count[output]>0` (or `0<count output` to avoid square brackets)